src/components/DayPickerRangeController.jsx

```diff
@@ -457,51 +475,76 @@ export default class DayPickerRangeController extends React.Component {
       endDate,
       focusedInput,
       minimumNights,
+      range,
     } = this.props;
     const { hoverDate, visibleDays } = this.state;
+    let dateRange = null;
 
     if (focusedInput) {
       let modifiers = {};
-      modifiers = this.deleteModifier(modifiers, hoverDate, 'hovered');
-      modifiers = this.addModifier(modifiers, day, 'hovered');
 
-      if (startDate && !endDate && focusedInput === END_DATE) {
-        if (isAfterDay(hoverDate, startDate)) {
-          const endSpan = hoverDate.clone().add(1, 'day');
-          modifiers = this.deleteModifierFromRange(modifiers, startDate, endSpan, 'hovered-span');
-        }
+      if (range && (range.before || range.after)) {
+        const start = range.before ? range.before(day.clone()) : day.clone();
```

Comment:
this code seems repeated; can it live in somewhere shared?

Reply:
👍